src/toil/worker.py

```diff
@@ -415,7 +419,7 @@ def badWorker():
     #so safe to test if they completed okay
     ########################################## 
     
-    if Job.FileStore._terminateEvent.isSet():
+    if FileStore._terminateEvent.isSet():
         jobWrapper = jobStore.load(jobStoreID)
         jobWrapper.setupJobAfterFailure(config)
         workerFailed = True
```

Comment:
Wow! This is a huge PR. I think I'd like to see test stats on how well it works before we merge, just in case, but my cursory read didn't throw up any big red flags. I definitely think we should remove my first version of caching, once you've done a comparison.